File: sentimental-credit/credit.py
def valid(num):
    sum = 0
    length = len(num)
    while length > 1:
        product = 2 * num[length - 2]
        for digit in str(product):
            sum += int(digit)
        length -= 2
    length = len(num)
    while length > 0:
        sum += num[length - 1]
        length -= 2
    if sum % 10 == 0:
        return True
    else:
        return False


def main():
    while True:
        try:
            number = int(input("Number: "))
            if number > 0:
                break
        except ValueError:
            print(end="")

    num = [int(x) for x in str(number)]

    if len(num) == 15:
        if num[0] == 3 and (num[1] == 4 or num[1] == 7):
            if valid(num) == True:
                print("AMEX")
                return
            else:
                print("INVALID")
                return
        else:
            print("INVALID")
            return

    elif len(num) == 16:
        if num[0] == 5 and (num[1] >= 1 and num[1] <= 5):
            if valid(num) == True:
                print("MASTERCARD")
                return
            else:
                print("INVALID")
                return

        elif num[0] == 4:
            if valid(num) == True:
                print("VISA")
                return
            else:
                print("INVALID")
                return

        else:
            print("INVALID")
            return

    elif len(num) == 13:
        if num[0] == 4:
            if valid(num) == True:
                print("VISA")
                return
            else:
                print("INVALID")
                return
        else:
            print("INVALID")
            return

    else:
        print("INVALID")
        return


main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>


bool valid(long n);


int main(void)
{

    long cardNumber = get_long("Number: ");

    if(cardNumber / 100000000000000 != 0 && ((cardNumber / 100000000000000 == 3 && (cardNumber / 10000000000000) % 10 == 4) || (cardNumber / 100000000000000 == 3 && (cardNumber / 10000000000000) % 10 == 7)))
    {

        if(valid(cardNumber) == true)
        {
            printf("AMEX\n");
        }
        else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
    }
    else if(cardNumber / 1000000000000000 == 4 && cardNumber / 10000000000000000 == 0)
    {
        if(valid(cardNumber) == true)
        {
            printf("VISA\n");
        }
        else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
    }
    else if(cardNumber / 1000000000000 == 4 && cardNumber / 10000000000000 == 0)
    {
        if(valid(cardNumber) == true)
        {
            printf("VISA\n");
        }
        else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
    }
    else if(cardNumber / 1000000000000000 == 5 && cardNumber / 10000000000000000 == 0)
    {
        if((cardNumber / 100000000000000) % 10 == 1)
        {
            if(valid(cardNumber) == true)
            {
            printf("MASTERCARD\n");
            }
            else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
        }
        else if((cardNumber / 100000000000000) % 10 == 2)
        {
            if(valid(cardNumber) == true)
            {
            printf("MASTERCARD\n");
            }
            else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
        }
        else if((cardNumber / 100000000000000) % 10 == 3)
        {
            if(valid(cardNumber) == true)
            {
            printf("MASTERCARD\n");
            }
            else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
        }
        else if((cardNumber / 100000000000000) % 10 == 4)
        {
            if(valid(cardNumber) == true)
            {
            printf("MASTERCARD\n");
            }
            else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }
        }
        else if((cardNumber / 100000000000000) % 10 == 5)
        {
            if(valid(cardNumber) == true)
            {
            printf("MASTERCARD\n");
            }
            else if(valid(cardNumber) == false)
            {
                printf("INVALID\n");
            }

        }
        else
        {
            printf("INVALID\n");
        }
    }

    else
    {
        printf("INVALID\n");
    }


}



bool valid(long n)
{
    long number;
    long sum = 0;

    for(long i = 10; i <= 1000000000000000; i *= 100)
    {
        number = ((n / i) % 10) * 2;

        sum = sum + number / 10 + number % 10;

    }
    for(long i = 1; i <= 1000000000000000; i*= 100)
    {
        number = (n / i) % 10;
        sum = sum + number % 10;
    }

    if(sum % 10 == 0)
    {
        return true;
    }
    else if(sum % 10 != 0)
    {
        return false;
    }
    else
    {
        return false;
    }


}

